_rules/SVG-image-has-name.md

```diff
@@ -0,0 +1,201 @@
+
+```html
+<svg xmlns="https://www.w3.org/2000/svg" width="600" height="400" viewBox="200 0 400 400" role="graphics-document">
+	<title>3 shapes</title>
+	<g role="graphics-object" aria-label="Blue rectangle" transform="translate(100,100)">
+        <desc>Large blue rectangle with two other shapes inside it</desc>
+		<rect fill="blue" role="graphics-object" aria-label=" large blue rectangle" width="200" height="100" y="-100" />
+		<g role="graphics-object" aria-label="small yellow rectangle in larger green rectangle" transform="translate(30,-90)">
+			<rect fill="green" width="50" height="90"/>
+			<rect fill="yellow" x="5" y="5" width="40" height="30" />
+			</g>
+		 </g>
+	</svg>
+```
+
+### Failed
+
+#### Failed example 1
+
+The `<svg>` element has a role of `img` but has no accessible name.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" role="img" width="100" height="100">
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+
+#### Failed example 2
+
+The `img` role includes the `<svg>` element in the accessibility tree but the element has an empty accessible name `title` attribute.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100" role="img">
+	<title> </title>
+  	<circle role="none" cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+#### Failed example 3
+
+The `svg` `<circle>` elements are included in the accessibility tree but do not have accessible names.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" role="graphics-document">
+	<title>Yellow shapes</title>
+	<circle role="graphics-symbol" cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+	<circle role="graphics-symbol" cx="150" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+	<circle role="graphics-symbol" cx="250" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+#### Failed example 4
+
+The `<g>` elements are included in the accessibility tree with the `graphics-object` role but only one of the elements has an accessible name.
+
+```html
+<svg xmlns="https://www.w3.org/2000/svg" width="600" height="400" viewBox="200 0 400 400" role="graphics-document">
+	<title>3 shapes</title>
+	<g role="graphics-object" aria-label="Blue rectangle" transform="translate(100,100)">
+        <desc>Large blue rectangle with two other shapes inside it</desc>
+		<rect fill="blue" role="graphics-object" aria-label=" large blue rectangle" width="200" height="100" y="-100" />
+		<g role="graphics-object" transform="translate(30,-90)">
+			<rect fill="green" width="50" height="90"/>
+			<rect fill="yellow" x="5" y="5" width="40" height="30" />
+			</g>
+		 </g>
+	</svg>
+```
+
+### Inapplicable
+
+#### Inapplicable example 1
+
+The `svg` element does not have a role of `img` `graphics-document` `graphics-object` or `graphics-symbol`. 
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100">
```

Comment:
inapplicable or decorative? In my Imaginary Bot :-) Bot: "Is this a decorative non-text element?" Me: "Yes" (if yes, so is acceptable, don't do nothing); Me: "No" (if no, so it needs an a11y name). In PT Public Administration's websites this is the most common situation. So, like is inapplicable, no errors, no warnings... is perfect!

Reply:
How to consider the situation of a link with only a svg element by content: 
<a ...><svg></svg></a>

Reply:
I think both of these questions need to be answered by other rules. 
A rule to ask "should content should be marked as decorative?". 
And separately a link with only an svg element for content should be covered by a rule testing link content.

Reply:
@kasperisager says it has an implicit role of `graphics-document` and "Elements which are never directly rendered to screen, nor represented by an interactive region in a graphic, do not need a corresponding accessible object." (https://www.w3.org/TR/svg-aam-1.0/#mapping_general)